src/Type/Constant/ConstantArrayType.php

```diff
@@ -468,7 +468,19 @@ public function isCallable(): TrinaryLogic
 			$typeAndMethods,
 		);
 
-		return TrinaryLogic::createYes()->and(...$results);
+		$isCallable = TrinaryLogic::createYes()->and(...$results);
+		if ($isCallable->yes()) {
+			$callableArray = $this->getClassOrObjectAndMethods();
```

Comment:
This is ineffective. This method was already called once inside `findTypeAndMethodNames` above. It's a bit of a mess right now.

I just made sure this kind of mistake isn't possible: https://github.com/phpstan/phpstan-src/commit/30b9cd86c523b59739b609dc4e325d3275d71188

So the fix should probably be done inside `findTypeAndMethodNames` but I'm not sure.

Reply:
> So the fix should probably be done inside findTypeAndMethodNames but I'm not sure.

Not sure too. Depends of the purpose of the method.

Currently the `findTypeAndMethodNames` returns all the possible callable with at least a Maybe certainty. Cf https://github.com/phpstan/phpstan-src/pull/3637#issuecomment-2485308668

Currently it's not possible to create
```
ConstantArrayTypeAndMethod::createConcrete($type, $method->getValue(), TrinaryLogic::createNo());
```

So if you have a string 'foo'|'bar' and only 'foo' is a callable name, if it will returns 'foo'.
But it doesn't mean that 'foo'|'bar' is always a callable.

That's why I added the logic in isCallable to check the count of possible callable didn't drop (which mean some value has No as certainty).